### PR TITLE
`azurerm_cosmosdb_mongo_collection` - correctly read the `_id` index back when mongo 3.6

### PR DESCRIPTION
**Pull request by yupwei68**
Fix https://github.com/terraform-providers/terraform-provider-azurerm/issues/8660

=== RUN   TestAccAzureRMCosmosDbMongoCollection_basic
=== PAUSE TestAccAzureRMCosmosDbMongoCollection_basic
=== CONT  TestAccAzureRMCosmosDbMongoCollection_basic
--- PASS: TestAccAzureRMCosmosDbMongoCollection_basic (1811.27s)
=== RUN   TestAccAzureRMCosmosDbMongoCollection_complete
=== PAUSE TestAccAzureRMCosmosDbMongoCollection_complete
=== CONT  TestAccAzureRMCosmosDbMongoCollection_complete
--- PASS: TestAccAzureRMCosmosDbMongoCollection_complete (1819.16s)
=== RUN   TestAccAzureRMCosmosDbMongoCollection_update
=== PAUSE TestAccAzureRMCosmosDbMongoCollection_update
=== CONT  TestAccAzureRMCosmosDbMongoCollection_update
--- PASS: TestAccAzureRMCosmosDbMongoCollection_update (2097.46s)
=== RUN   TestAccAzureRMCosmosDbMongoCollection_throughput
=== PAUSE TestAccAzureRMCosmosDbMongoCollection_throughput
=== CONT  TestAccAzureRMCosmosDbMongoCollection_throughput
--- PASS: TestAccAzureRMCosmosDbMongoCollection_throughput (2052.64s)
=== RUN   TestAccAzureRMCosmosDbMongoCollection_withIndex
=== PAUSE TestAccAzureRMCosmosDbMongoCollection_withIndex
=== CONT  TestAccAzureRMCosmosDbMongoCollection_withIndex
--- PASS: TestAccAzureRMCosmosDbMongoCollection_withIndex (1824.25s)
=== RUN   TestAccAzureRMCosmosDbMongoCollection_autoscale
=== PAUSE TestAccAzureRMCosmosDbMongoCollection_autoscale
=== CONT  TestAccAzureRMCosmosDbMongoCollection_autoscale
--- PASS: TestAccAzureRMCosmosDbMongoCollection_autoscale (2234.07s)
=== RUN   TestAccAzureRMCosmosDbMongoCollection_ver36
=== PAUSE TestAccAzureRMCosmosDbMongoCollection_ver36
=== CONT  TestAccAzureRMCosmosDbMongoCollection_ver36
--- PASS: TestAccAzureRMCosmosDbMongoCollection_ver36 (1685.88s)

### CODE REVIEW

**Review by jackofallops — CHANGES_REQUESTED** [review]
Hi @yupwei68 
Thanks for this PR. I'm afraid I think this needs some more thought.  The linked issue, and indeed the older related issue #8144, indicate that the `_id` index should always be present if custom indexes are specified after the breaking change in behaviour for the API.  Native MongoDB creates this, but CosmosDB Mongo does not.

In a 3.6 Collection the `_id` index _is_ returned by the API (I've not tried the 3.2 version, but I suspect it's there also)
```
{
  "id": "/subscriptions/xxx/resourceGroups/xxx/providers/Microsoft.DocumentDB/databaseAccounts/dev-ca/apis/mongodb/databases/dev-cosmosmongo/collections/dev-cosmos-collection",
  "type": "Microsoft.DocumentDB/databaseAccounts/apis/databases/collections",
  "name": "dev-cosmos-collection",
  "properties": {
    "id": "dev-cosmos-collection",
    "_rid": "CsMXAJWjA5o=",
    "_etag": "\"00003304-0000-0d00-0000-5f9956260000\"",
    "_ts": 1603884582,
    "indexes": [
      {
        "key": {
          "keys": [
            "_id"
          ]
        },
        "options": {}
      },
      {
        "key": {
          "keys": [
            "_ts"
          ]
        },
...
```
The service defaults to 3.2 when created by the API (The portal defaults to 3.6). This is changed by sending the `EnableMongo` value in the `capabilities` list on the account. (this is definitely not obvious), so your test doesn't exercise the correct path I'm afraid.  

In my comments on 8144 for potential solutions, there were a couple approaches we could take. We document that the user needs to specify the `_id` index when creating a collection, or we add it in the code if indexes are specified but that one is missing. Ideally the service shouldn't need us to do any of this. Native MongoDB creates that [index automatically when a new collection is created](https://docs.mongodb.com/v3.6/indexes/#default-id-index), so the ideal situation would be that the service team match Native behaviour that users would expect from a MongoDB Service.

**Comment by yupwei68** [comment]
Thanks @jackofallops for more context and information. Then I think the problem exists in that we have set the `index` "_id" to `system_index`. I've done the corresponding changes. All tests have passed. Please continue reviewing.

=== RUN   TestAccAzureRMCosmosDbMongoCollection_basic
=== PAUSE TestAccAzureRMCosmosDbMongoCollection_basic
=== CONT  TestAccAzureRMCosmosDbMongoCollection_basic
--- PASS: TestAccAzureRMCosmosDbMongoCollection_basic (1637.67s)
=== RUN   TestAccAzureRMCosmosDbMongoCollection_complete
=== PAUSE TestAccAzureRMCosmosDbMongoCollection_complete
=== CONT  TestAccAzureRMCosmosDbMongoCollection_complete
--- PASS: TestAccAzureRMCosmosDbMongoCollection_complete (1643.92s)
=== RUN   TestAccAzureRMCosmosDbMongoCollection_update
=== PAUSE TestAccAzureRMCosmosDbMongoCollection_update
=== CONT  TestAccAzureRMCosmosDbMongoCollection_update
--- PASS: TestAccAzureRMCosmosDbMongoCollection_update (1953.35s)
=== RUN   TestAccAzureRMCosmosDbMongoCollection_throughput
=== PAUSE TestAccAzureRMCosmosDbMongoCollection_throughput
=== CONT  TestAccAzureRMCosmosDbMongoCollection_throughput
--- PASS: TestAccAzureRMCosmosDbMongoCollection_throughput (2160.28s)
=== RUN   TestAccAzureRMCosmosDbMongoCollection_withIndex
=== PAUSE TestAccAzureRMCosmosDbMongoCollection_withIndex
=== CONT  TestAccAzureRMCosmosDbMongoCollection_withIndex
--- PASS: TestAccAzureRMCosmosDbMongoCollection_withIndex (1639.39s)
=== RUN   TestAccAzureRMCosmosDbMongoCollection_autoscale
=== PAUSE TestAccAzureRMCosmosDbMongoCollection_autoscale
=== CONT  TestAccAzureRMCosmosDbMongoCollection_autoscale
--- PASS: TestAccAzureRMCosmosDbMongoCollection_autoscale (2013.61s)
=== RUN   TestAccAzureRMCosmosDbMongoCollection_ver36
=== PAUSE TestAccAzureRMCosmosDbMongoCollection_ver36
=== CONT  TestAccAzureRMCosmosDbMongoCollection_ver36
--- PASS: TestAccAzureRMCosmosDbMongoCollection_ver36 (1787.49s)

**Comment by LaurentLesle** [comment]
@yupwei68 @jackofallops  do you think you can complete this PR soon? Looks like an issue pending on this PR https://github.com/terraform-providers/terraform-provider-azurerm/issues/8660

File: azurerm/internal/services/cosmos/cosmosdb_mongo_collection_resource.go [hunk 1]
```diff
@@ -313,8 +320,16 @@ func resourceArmCosmosDbMongoCollectionRead(d *schema.ResourceData, meta interfa
 			for k := range res.ShardKey {
 				d.Set("shard_key", k)
 			}
+			accountIsVersion36 := false
+			if accProps := accResp.DatabaseAccountGetProperties; accProps != nil {
+				for _, v := range *accProps.Capabilities {
```
Comment (katbyte) [inline azurerm/internal/services/cosmos/cosmosdb_mongo_collection_resource.go:325]: We should add a nil check here

File: azurerm/internal/services/cosmos/cosmosdb_mongo_collection_resource.go [hunk 2]
```diff
@@ -313,8 +320,16 @@ func resourceArmCosmosDbMongoCollectionRead(d *schema.ResourceData, meta interfa
...
+					if *v.Name == "EnableMongo" {
```
Comment (katbyte) [inline azurerm/internal/services/cosmos/cosmosdb_mongo_collection_resource.go:326]: We should add a nil check here

**Review by katbyte — CHANGES_REQUESTED** [review]
Thanks @yupwei68 - couple comments but otherwise looks good

**Comment by yupwei68** [comment]
@katbyte Thanks for your comments. Please continue reviewing.

**Review by katbyte — APPROVED** [review]
Thanks @yupwei68 - LGTM 👍

**Pull request merged**